### ISSUE TITLE
Problems with the new version: order of the options and looping

### ISSUE THREAD

**Issue by crawler-IM** [openpreserve/pagelyzer#5]
Hi, 
To run the tool correctly we should first define the JAVA_HOME :
export JAVA_HOME=/usr/lib/jvm/java-6-openjdk/

if not, we got this exception
/usr/bin/pagelyzer_changedetection:90:in `load': can't create Java VM (RuntimeError)
    from /usr/bin/pagelyzer_changedetection:90:in`get_java_instance'
    from /usr/bin/pagelyzer_changedetection:112:in `<main>'

Then we can run it, in our case we use the command the pagelyzer_changedetection in order to compare between two web pages.
1- We should note, first , that the execution of the tool depends on the order of its options, example :

A-
$ pagelyzer_changedetection --url1=http://webarchive.nationalarchives.gov.uk/20100824180635/http://yourfreedom.hmg.gov.uk/ --url2=http://im1c6.internetmemory.org/tna/20100824180635/http://yourfreedom.hmg.gov.uk/ --headless --output-folder=./file-0-line/ --url-archive
Headless mode
Notice: using 'firefox' as default browser
Type: images. Using: /etc/pagelyzer_ruby/ex_images.xml file

Connection not possible with firefox
WARNING: Is firefox installed in your system?
.....

B-
$ pagelyzer_changedetection --url1=http://webarchive.nationalarchives.gov.uk/20100824180635/http://yourfreedom.hmg.gov.uk/ --url2=http://im1c6.internetmemory.org/tna/20100824180635/http://yourfreedom.hmg.gov.uk/ --headless --url-archive --output-folder=./file-0-line-0
Headless mode
Notice: using 'firefox' as default browser
Type: images. Using: /etc/pagelyzer_ruby/ex_images.xml file

Timeout: 30secs
Capturing http://webarchive.nationalarchives.gov.uk/20100824180635/http://yourfreedom.hmg.gov.uk/ with local firefox
Getting screenshot
Waiting page to finish loading...
done.
Timeout: 30secs
Capturing http://im1c6.internetmemory.org/tna/20100824180635/http://yourfreedom.hmg.gov.uk/ with local firefox
Getting screenshot
Waiting page to finish loading...
done.
0.00687000358334055
0.0166589550943836
0.0031247369478646305
0.00896583116319136
Distance between the two web-pages:: -0.5490586292655766
Processed in 71.005285768secs
Browser firefox closed

2- The second note is about the option --output-folder : if we don't create the output folder before,  (in the last version, if the folder does not exist, it will be created) :
$ pagelyzer_changedetection --url1=http://webarchive.nationalarchives.gov.uk/20100824180635/http://yourfreedom.hmg.gov.uk/ --url2=http://im1c6.internetmemory.org/tna/20100824180635/http://yourfreedom.hmg.gov.uk/ --headless --url-archive --output-folder=./file-0-line-0 
Headless mode
Notice: using 'firefox' as default browser
Type: images. Using: /etc/pagelyzer_ruby/ex_images.xml file

Connection not possible with firefox
WARNING: Is firefox installed in your system?
....

In other side, after using the option with an existing folder, we dont find the files used in the comparaison (.html, .png and marcalizer folder).

These were some examples of exécution while testing the tool, but our use case is run pagelyzer_changedetection in a loop, comparing many web pages. Using the command with needed options (in the right order) we note different result for the same command executed, plesae find the command runned and the result in order :

pagelyzer_changedetection --url1=http://webarchive.nationalarchives.gov.uk/20100824180635/http://yourfreedom.hmg.gov.uk/ --url2=http://im1c6.internetmemory.org/tna/20100824180635/http://yourfreedom.hmg.gov.uk/ --headless  --url-archive --output-folder=./file-0-line-0
pagelyzer_changedetection --url1=http://webarchive.nationalarchives.gov.uk/20120406035308/http://www.coi.gov.uk/ --url2=http://im1c6.internetmemory.org/tna/20120406035308/http://www.coi.gov.uk/ --headless  --url-archive --output-folder=./file-0-line-1
pagelyzer_changedetection --url1=http://webarchive.nationalarchives.gov.uk/20110809101133/http://nsonline.org.uk/site_intelligence/selectors/page?H --url2=http://im1c6.internetmemory.org/tna/20110809101133/http://nsonline.org.uk/site_intelligence/selectors/page?H --headless  --url-archive --output-folder=./file-0-line-2

$ cat log.txt 
Tue Apr 23 11:12:27 CEST 2013
Headless mode
Notice: using 'firefox' as default browser
Type: images. Using: /etc/pagelyzer_ruby/ex_images.xml file
Connection not possible with firefox
WARNING: Is firefox installed in your system?
Try with the --browser=your_installed_browser
USAGE: pagelyzer_changedetection [--url1=URL --url2=URL | --urls=FILE --output-file=FILE] [conf=CONF_FILE] [--doc=(1..10)] [--output-folder=FOLDER] [--browser=BROWSER_CODE | --browser1=BROWSER_CODE --browser2=BROWSER_CODE] [--verbose] --type=[images|structure|hybrid] [--url-archive]
This tool aims integrates all the change detection and segmentation tools
Help:
 type = hybrid | webshot
Browsers code are the same as defined in selenium. For instance:
- firefox
- chrome
- iexploreproxy
- safariproxy
- opera

For the input URL file it expects the following syntax of each line:
-URL1 URL2
Timeout: 30secs
there were a problem in the capture of pages
ERROR: can't process these urls:
http://webarchive.nationalarchives.gov.uk/20100824180635/http://yourfreedom.hmg.gov.uk/
http://im1c6.internetmemory.org/tna/20100824180635/http://yourfreedom.hmg.gov.uk/
Tue Apr 23 11:14:39 CEST 2013  

---

Tue Apr 23 11:14:39 CEST 2013
Headless mode
Notice: using 'firefox' as default browser
Type: images. Using: /etc/pagelyzer_ruby/ex_images.xml file
0.012651629581915857
0.021330379109104593
0.022870095264484743
0.02865328747913551
Distance between the two web-pages:: 0.6305194657627191
Timeout: 30secs
Capturing http://webarchive.nationalarchives.gov.uk/20120406035308/http://www.coi.gov.uk/ with local firefox
Getting screenshot
Waiting page to finish loading...
done.
Timeout: 30secs
Capturing http://im1c6.internetmemory.org/tna/20120406035308/http://www.coi.gov.uk/ with local firefox
Getting screenshot
Waiting page to finish loading...
done.
Processed in 54.306332047secs
Browser firefox closed
Tue Apr 23 11:16:07 CEST 2013  

---

Tue Apr 23 11:16:07 CEST 2013
Headless mode
Notice: using 'firefox' as default browser
Type: images. Using: /etc/pagelyzer_ruby/ex_images.xml file
0.0
0.0
0.0
0.0
Distance between the two web-pages:: 1.0006352506978375
Timeout: 30secs
Capturing http://webarchive.nationalarchives.gov.uk/20110809101133/http://nsonline.org.uk/site_intelligence/selectors/page?H with local firefox
Getting screenshot
Waiting page to finish loading...
done.
Timeout: 30secs
Capturing http://im1c6.internetmemory.org/tna/20110809101133/http://nsonline.org.uk/site_intelligence/selectors/page?H with local firefox
Getting screenshot
Waiting page to finish loading...
done.
Processed in 12.933806415secs
Browser firefox closed
Tue Apr 23 11:17:16 CEST 2013  

---

Tue Apr 23 11:17:16 CEST 2013
Headless mode
Notice: using 'firefox' as default browser
Type: images. Using: /etc/pagelyzer_ruby/ex_images.xml file
Timeout: 30secs
Capturing http://webarchive.nationalarchives.gov.uk/20100307172104/http://ofsted.gov.uk/ofsted-home/about-us/faqs with local firefox
Getting screenshot
Waiting page to finish loading...
done.
Timeout: 30secs
Capturing http://im1c6.internetmemory.org/tna/20100307172104/http://ofsted.gov.uk/ofsted-home/about-us/faqs with local firefox
ERROR: Page load timeout execution expired
there were a problem in the capture of pages
ERROR: can't process these urls:
http://webarchive.nationalarchives.gov.uk/20100307172104/http://ofsted.gov.uk/ofsted-home/about-us/faqs
http://im1c6.internetmemory.org/tna/20100307172104/http://ofsted.gov.uk/ofsted-home/about-us/faqs
Browser firefox closed
Tue Apr 23 11:18:22 CEST 2013    

---

As mentionned above, we found 4 differents results : 
Failure : Connection not possible with firefox
Success : Distance between the two web-pages:: 0.6305194657627191
Success (I suppose) : Distance between the two web-pages:: 1.0006352506978375
Failure : ERROR: Page load timeout execution expired (I think it can be fixed with --timeout option)

To summarize, the debian package is easy to install (with dpkg command) and the option --headless is well integrated for the systems without GUI. But in other side, the new version does not keep the snapshots done, the order of the option appears to be important and it seems to be not stable while using in a loop.

Thanks.

**Comment by ZeynepP** [comment]
1- We are not able to reproduce the error for instant. We are on it.

2- "In other side, after using the option with an existing folder, we dont find the files used in the comparaison (.html, .png and marcalizer folder)." We were agreed about using streams insted of files. So,  we don't write anymore anything to the folders except the results. 

3- We intoduced a new --urls  that takes a file with a structure like :

url1 \space url2 \n

so you don't need anymore to make a loop each time for different pairs of urls pagelyzer will handle it internally. 

Bests 
ZP

**Comment by asanoja** [comment]
Hi, the order of the two parameters has been checked and we has not been able to reproduce it. However we checked this part of the code to be sure that either this kind of things do not happens or to produce a message error appropriate. It can be presumably due to issues natural in the navigation cycle and the response of the browser for each url.

In the other hand the issue with the output file has been solved. It will create both folder and file as needed.

As in precedent message the loop is not needed, only the parameter --urls is needed.
